### ISSUE TITLE
Support `use enum enumName { const1, const2, const3 };`

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#18826]
You know how, sometimes, you just want to declare and use an `enum` without
the fully qualified form, and without typing a separate `use` statement, as though
you were a C programmer again?  What we supported:

```chapel
use enum color {red, green, blue};
```

as a shorthand for:

```chapel
enum color {red, green, blue};
use color;
```

?

**Comment by lydia-duncan** [comment]
Wouldn't that mean you could never get access to the enum without unqualified access to its constants?  To be fair, that still happens with a use declared at the same scope as the enum, but at least we're not explicitly encouraging it, just allowing it.

**Comment by bradcray** [comment]
> Wouldn't that mean you could never get access to the enum without unqualified access to its constants?

That's correct.

> but at least we're not explicitly encouraging it, just allowing it.

I guess I don't see making something easier the same as encouraging it.  If we had a "standard library meets coding standards linter", for example, I would expect it to consider this a bad practice.  But I think it's a perfectly fine practice when sketching out some code quickly (where the two statement form just feels clunky to me).

By analogy, it feels a little bit like if we required public uses and imports to be specified using a separate statement:

```
use M;  // private use of M, like always/normal
reexport M;  // imagine this would be required to make it public rather than `public use M;`
```

**Comment by Nirhar** [comment]
Hey guys, I would like to take up this issue!
I am new to exploring chapel's codebase, but from whatever little that I explored, I think this can be solved by adding one line in the parser bison file that means:
`use_statement --> TUSE enum_decl_stmt`.
Do I have to consider any semantic issues that may arise?

Also could this issue be assigned to me?

**Comment by lydia-duncan** [comment]
I'm open to you attempting it, but would really prefer more eyes take a look at the proposal before committing to it being implemented

**Comment by bradcray** [comment]
I agree with Lydia, @Nirhar:  It would be fine to try implementing this, but we can't merge it (or even guarantee that we'll make the change) until there's been a design discussion to make sure that the core set of users and developers don't object to it since it's a language change.  See https://chapel-lang.org/docs/main/developer/bestPractices/ContributorInfo.html#design for details on our design process.

Also, note that while your change will likely let the capability parse, additional changes will be necessary to cause it to work as intended.  And we'll need to update the documentation as well.

**Comment by Nirhar** [comment]
Okay sure, let me know if you think it must be worked on! Would be glad to contribute!

**Comment by lydia-duncan** [comment]
I think it'd be fine for you to lead an email discussion if you felt up for it, but since we're so close to the winter holidays we'd probably want to wait on making any decisions until the second week of January, to give people a chance to see and comment on the idea.  And it's also okay to not feel comfortable leading a discussion

**Comment by bradcray** [comment]
On Friday, I edited the OP here to wonder about the public/private nature of `use enum` if we adopted this proposal, based on some discussion on https://github.com/chapel-lang/chapel/pull/19306#discussion_r819995043.
